app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -777,53 +779,65 @@ class HomeFragment : Fragment() {
             context,
             onItemTapped = {
                 when (it) {
-                    HomeMenu.Item.Settings -> {
+                    is Item.Settings -> {
```

Comment:
We only need to add the `is` for ~`SyncAccount`~ (see other comment about remove this) and `DesktopMode`, looking at this patch, since those are the only non-`object` classes that need to be type checked, the others are singleton objects; only one will ever exist.

Reply:
I prefer this for consistency